dali/kernels/slice/slice_hwc2chw_normalize_gpu.cu

```diff
@@ -0,0 +1,532 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <tuple>
+#include <type_traits>
+#include "dali/core/backend_tags.h"
+#include "dali/core/convert.h"
+#include "dali/core/error_handling.h"
+#include "dali/core/permute.h"
+#include "dali/core/static_switch.h"
+#include "dali/kernels/slice/slice_hwc2chw_normalize_gpu.h"
+
+namespace dali {
+namespace kernels {
+
+namespace slice_flip_normalize {
+
+template <typename Out, typename In>
+struct Hwc2ChwSampleDesc {
+  Out *__restrict__ out;
+  const In *__restrict__ in;
+  const float *__restrict__ norm_add;
+  const float *__restrict__ norm_mul;
+  const void *__restrict__ fill_values;
+
+  uint32_t first_block;
+  int64_t sample_size;
+
+  // Dimensions of the output
+  int H, W, C;
+  // Dimensions of the input (relevant to input stride)
+  int input_W, input_C;
+
+  bool flip_x;
+};
+
+template <typename Out, typename In>
+inline __device__ uint32_t FindSampleIdx(const Hwc2ChwSampleDesc<Out, In> *samples,
+                                         uint32_t num_samples) {
+  uint32_t i = 0;
+  for (uint32_t jump = (1 << (32 - __clz(num_samples) - 1)); jump; jump >>= 1) {
+    if (i + jump < num_samples && samples[i + jump].first_block <= blockIdx.x)
+      i += jump;
+  }
+  return i;
```

Comment:
In NCU I see a slight drop in timings (this can be measurement error, but it is consistent in one direction), and increase in the number of issues detected. I also don't see any improvement in the pipeline mircobenchmark that I'm using, so I will leave it as is.